bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/GraalJSScriptEngineFactory.java

```diff
@@ -32,13 +30,24 @@
 @Component(service = ScriptEngineFactory.class)
 public final class GraalJSScriptEngineFactory implements ScriptEngineFactory {
 
+    public static final String MIME_TYPE = "application/javascript;version=ECMAScript-2021";
+
     @Override
     public List<String> getScriptTypes() {
         List<String> scriptTypes = new ArrayList<>();
-        GraalJSEngineFactory graalJSEngineFactory = new GraalJSEngineFactory();
 
-        scriptTypes.addAll(graalJSEngineFactory.getMimeTypes());
-        scriptTypes.addAll(graalJSEngineFactory.getExtensions());
+        /*
+         * Whilst we run in parallel with Nashorn, we use a custom mime-type to avoid
+         * disrupting Nashorn scripts. When Nashorn is removed, we take over the standard
+         * JS runtime.
+         */
+
+        // GraalJSEngineFactory graalJSEngineFactory = new GraalJSEngineFactory();
```

Comment:
They are left there as a reminder of what the code should be when we eventually remove nashorn in a future release.

Reply:
@digitaldan  is correct. They are commented out now to allow this JS addon to coexist with Nashorn. At the time we remove Nashorn, we uncomment these lines and this will take over Nashorn scripts.

If there is another process to do this though, I'm happy to use it.

Reply:
Alright. Keep it in place.